### PR TITLE
[DO_NOT_MERGE] report proper error for unrecognized source protocol

### PR DESCRIPTION
**Pull request by bparees**
fixes https://bugzilla.redhat.com/show_bug.cgi?id=1190106
(once it makes it to origin)

### CODE REVIEW

**Comment by bparees** [comment]
@soltysh please review.

**Comment by bparees** [comment]
[test]

**Comment by soltysh** [comment]
LGTM, but can you please add test case to cover this situation? We've should work on our testing suite in sti, I'm not too happy with the test coverage in certain modules.

**Comment by bparees** [comment]
sure.

**Comment by soltysh** [comment]
Cool, you're gonna have to fix the test cases as well.

**Comment by bparees** [comment]
so i see :)

**Comment by bparees** [comment]
i've realized this is totally wrong :(    file:// is supposed to be treated as a valid git clone spec.

**Comment by bparees** [comment]
@soltysh still needs new tests but at least i think the code is right now :/

File: pkg/test/fs.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package test
 import (
 	"bytes"
 	"errors"
+	"fmt"
```
Comment (soltysh) [inline pkg/test/fs.go]: Remove this line and your good to go :wink:

File: pkg/errors/errors.go [hunk 2]
```diff
@@ -192,3 +193,14 @@ func NewContainerError(name string, code int, output string) error {
...
 }
+
+// NewSourcePathError returns a new error which indicates there was a problem
+// when accessing the source code from the local filesystem
+func NewSourcePathError(path string) error {
+	return Error{
+		Message:    fmt.Sprintf("Local filesystem source path does not exist: %s", path),
+		Details:    nil,
+		ErrorCode:  SourcePathError,
+		Suggestion: "check the source code path on hte local filesystem",
```
Comment (soltysh) [inline pkg/errors/errors.go]: s/on hte/on the

**Comment by soltysh** [comment]
Two small nits, but generally LGTM.

File: pkg/errors/errors.go (hunk 2 above)
Comment (bparees) [inline pkg/errors/errors.go]: no more coding at 1am for me :(

**Comment by bparees** [comment]
thanks @soltysh, nits fixed but integration test is still failing and i still want to add test cases.

File: pkg/errors/errors.go (hunk 2 above)
Comment (soltysh) [inline pkg/errors/errors.go]: I know the pain, I'm promising myself to go to bad at reasonable time this week, but too much to do and so not much time. Do you know how to stretch a day to fit at least 48h in it :wink:

**Comment by soltysh** [comment]
Will wait for tests to pass and merge right away.

**Comment by soltysh** [comment]
Jenkins green, not waiting for travis [merge]

**Pull request merged**